rotkehlchen/db/drivers/client.py

```diff
@@ -192,31 +192,43 @@ def __init__(
         self.lastrowid: int = 0
         self.cursor_name = cursor_name
         if cursor_name is not None:
-            self._send(DbMethod.OPEN_CURSOR, cursor_name=cursor_name)
+            self._send(DBMethod.OPEN_CURSOR, cursor_name=cursor_name)
 
-    def _send(self, method: DbMethod, cursor_name: str | None = None, *args: Any, **kwargs: Any) -> None:  # noqa: E501
-        self.zmq_connection.send_pyobj(ZMQCallData(
-            connection_type=self.connection_type,
-            db_path=self.db_path,
-            cursor_name=cursor_name,
-            method=method,
-            args=args,
-            kwargs=kwargs,
-        ))
-        response: SerializedZMQReturnData | None = self.zmq_connection.recv_pyobj()
-        if response is None:
+    def _send(self, method: db_messages_pb2.DBMethod, cursor_name: str | None = None, *args: Any, **kwargs: Any) -> None:  # type: ignore[name-defined]  # pylint: disable=no-member  # noqa: E501
+        call_data = db_messages_pb2.ZMQCallData()  # type: ignore[attr-defined]  # pylint: disable=no-member
+        call_data.connection_type = self.connection_type.value
+        call_data.db_path = self.db_path
+        if cursor_name is not None:
+            call_data.cursor_name = cursor_name
+        call_data.method = method
+        call_data.args.extend([to_typed_data(data) for data in args])
+        for key, value in kwargs.items():
+            typed_data = to_typed_data(value)
+            call_data.kwargs[key].type = typed_data.type
+            if typed_data.WhichOneof('data') == 'bytes':
+                call_data.kwargs[key].bytes = typed_data.bytes
+            if typed_data.WhichOneof('data') == 'array':
+                call_data.kwargs[key].array.CopyFrom(typed_data.array)
+        self.zmq_connection.send(call_data.SerializeToString())
+        response_serialized = self.zmq_connection.recv()
+        if response_serialized == b'':
             return
-        result = ZMQReturnData.deserialize(response)
-        if result.error is not None:
-            raise result.error
-        if result.result is not None:
-            self.lastrowid = result.result['lastrowid']
-            self.rowcount = result.result['rowcount']
+        return_data = db_messages_pb2.ZMQReturnData()  # type: ignore[attr-defined]  # pylint: disable=no-member
+        return_data.ParseFromString(response_serialized)
+        if return_data.HasField('error'):
+            if return_data.error.name == DBError.UnknownError:
+                raise ValueError(f'Got unknown DB error with message {return_data.error.message}')
```

Comment:
Dont't raise ValueError... Isn't there something better? I would create a custom error for this